custom_components/ecoflow/binary_sensor.py

```diff
@@ -72,19 +72,19 @@ def __updated(self, data: dict[str, Any]):
 
     def _on_updated(self, data: dict[str, Any]):
         if "in_power" in data:
-            self._in_power = bool(data["in_power"])
+            self._in_power = data["in_power"]
         if "out_power" in data:
-            self._out_power = bool(data["out_power"])
+            self._out_power = data["out_power"]
         if "battery_level" in data:
             self._battery_level = data["battery_level"]
         if "battery_level_max" in data:
             self._battery_level_max = data["battery_level_max"]
 
         if not self._in_power:
             self._attr_is_on = False
-        elif self._battery_level is not None and self._battery_level_max is not None and self._battery_level_max < self._battery_level:
+        elif (self._battery_level is not None) and (self._battery_level_max is not None) and (self._battery_level_max < self._battery_level):
```

Comment:
What is this check supposed to do? I take it that max battery level will never be lower than current level?

Reply:
`battery_level_max` is a user-specified maximum charge limit. 
This condition is necessary because the user can intentionally set a limit lower than the current remaining charge.

Reply:
Ah, got it that's the custom max limit. That makes perfect sense then :+1: